source/MaterialXView/ShaderValidators/Osl/OslValidator.cpp

```diff
@@ -43,23 +44,111 @@ void OslValidator::initialize()
     }
 }
 
-void OslValidator::shadeOSL(const std::string& shaderName, const std::string& outputName)
+void OslValidator::renderOSL(const std::string& outputPath, const std::string& shaderName, const std::string& outputName)
+{
+    // If command options missing, skip testing.
+    if (_oslTestRenderExecutable.empty() || _oslIncludePathString.empty() || 
+        _oslTestRenderSceneTemplateFile.empty() || _oslUtilityOSOPath.empty())
+    {
+        return;
+    }
+
+    // Determine the shader path from output path and shader name
+    FilePath shaderFilePath(outputPath);
+    shaderFilePath = shaderFilePath / shaderName;
+    std::string shaderPath = shaderFilePath.asString();
+
+    // Set output image name. 
+    std::string outputFileName = shaderPath + ".testrender.png";
+
+    // Use a known error file name to check
+    std::string errorFile(shaderPath + "_render_errors.txt");
+    const std::string redirectString(" 2>&1");
+
+    // Read in scene template and replace "%shader%" 
+    // "%shader_output%" string siwht shader name and shader output
+    // respectively. Write to local file to use as input for rendering.
+    //
+    std::ifstream sceneTemplateStream(_oslTestRenderSceneTemplateFile);
+    std::string sceneTemplateString;
+    sceneTemplateString.assign(std::istreambuf_iterator<char>(sceneTemplateStream),
+        std::istreambuf_iterator<char>());
+
+    StringMap replacementMap;
```

Comment:
Token replacement to insert calling into shader to test and connecting the output of it to the constant output shader. Will need more work for surface shader support since the scene will differ.